include/seqan3/alignment/matrix/detail/trace_iterator.hpp

```diff
@@ -0,0 +1,234 @@
+     * \param[in] other The underlying matrix iterator.
+     */
+    template <TwoDimensionalMatrixIterator other_matrix_iter_t>
+    //!\cond
+        requires std::Constructible<matrix_iter_t, other_matrix_iter_t>
+    //!\endcond
+    constexpr trace_iterator(trace_iterator<other_matrix_iter_t> const other) noexcept :
+        trace_iterator{other.matrix_iter}
+    {}
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    //!\brief Returns the current traceback direction.
+    reference operator*() const noexcept
+    {
+        return cur_dir;
+    }
+
+    //!\brief Returns the current coordinate in two-dimensional space.
+    constexpr matrix_coordinate coordinate() const noexcept
+    {
+        return matrix_iter.coordinate();
+    }
+    //!\}
+
+    /*!\name Arithmetic operators
+     * \{
+     */
+    //!\brief Advances the iterator by one.
+    constexpr trace_iterator & operator++() noexcept
+    {
+        trace_directions old_dir = *matrix_iter;
+
+        assert(old_dir != trace_directions::none);
+
+        if (cur_dir == trace_directions::up)
+        {
+            matrix_iter -= matrix_offset{row_index_type{1}, column_index_type{0}};
+            // Set new traceback direction if last was up_open.
+            if (static_cast<bool>(old_dir & trace_directions::up_open))
+                set_trace_direction(); // Set the current trace_direction
+        }
+        else if (cur_dir == trace_directions::left)
+        {
+            matrix_iter -= matrix_offset{row_index_type{0}, column_index_type{1}};
+            // Set new traceback direction if last position was left_open.
+            if (static_cast<bool>(old_dir & trace_directions::left_open))
+                set_trace_direction(); // Set the current trace_direction
+        }
+        else
+        {
+            assert(cur_dir == trace_directions::diagonal);
+
+            matrix_iter -= matrix_offset{row_index_type{1}, column_index_type{1}};
+            set_trace_direction(); // Set the current trace_direction
+        }
+        return *this;
+    }
+
+    //!\brief Returns an iterator advanced by one.
+    constexpr trace_iterator operator++(int) noexcept
+    {
+        trace_iterator tmp{*this};
+        ++(*this);
+        return tmp;
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \{
+     */
+    //!\brief Returns `true` if both iterators are equal, `false` otherwise.
+    template <TwoDimensionalMatrixIterator other_matrix_iter_t>
+    //!\cond
+        requires std::Constructible<matrix_iter_t, other_matrix_iter_t> ||
+                 std::Constructible<other_matrix_iter_t, matrix_iter_t>
+    //!\endcond
+    constexpr bool operator==(trace_iterator<other_matrix_iter_t> const & rhs) const noexcept
+    {
+        return *matrix_iter == *rhs.matrix_iter;
+    }
+
+    //!\brief Returns `true` if the point-to-element is seqan3::detail::trace_directions::none.
```

Comment:
point-to-element? What does that mean?

Reply:
ok sit should be pointer-to-element. And it means what it says 🤓 . The element pointed to by the iterator